src/lightning/data/streaming/dataloader.py

```diff
@@ -386,7 +474,33 @@ def load_state_dict(self, obj: Dict[str, Any]) -> None:
             obj (Any): The state.
 
         """
-        if isinstance(self.dataset, (StreamingDataset, CombinedStreamingDataset)):
+        self.current_epoch = obj["current_epoch"]
+
+        if isinstance(self.dataset, StreamingDataset):
+            self._num_samples_yielded_streaming = obj["num_samples_yielded"]
+        else:
+            self._num_samples_yielded_combined = obj["num_samples_yielded"]
+
+        # Used to restart on the next DataLoader worker from the previous run.
+        self._latest_worker_idx = obj["latest_worker_idx"] + 1
+        self._worker_idx_iter = iter(self._worker_idx)
+        for _ in range(self._latest_worker_idx):
+            next(self._worker_idx_iter)
+
+        # Inform we are resuming
+        self.restore = True
```

Comment:
Don't we need to reset the variable to False somewhere?

Reply:
Yes, this is put back to False when the `__iter__` method of the StreamingDataLoader is finished. Added a comment.

Reply:
I don't see where this happens. iter doesn't set `self.restore= False` anywyere

Reply:
Right there: https://github.com/Lightning-AI/pytorch-lightning/pull/19326/files#diff-8b7ebda1bea12fa9fec6b7ffa6e0242ff1821a1f0868d2c60a16a02d5a0bbe93R408

Reply:
Added back and added a test ;) Thanks.